### PR TITLE
linter: new rule for declare(strict_types)

### PR DESCRIPTION
**Pull request by Hidanio**
- added new linter rule: declareStrictTypes can not be "0"
- added quickFix for declareStrictTypes
- added new linter rule: lack of declare(strict_types=1) section
-  added quickFix for lack of declare(strict_types=1) section
- added & updated tests
- updated doc
- new rules for .gitignore
- refactored warnings

### CODE REVIEW

File: docs/install.md [hunk 1]
```diff
@@ -73,13 +73,13 @@ noverify version
 
 You're done! Proceed to the [Getting started](/docs/getting_started.md) page.
 
-## With `go get`
+## With `go install`
```
Comment (i582) [inline docs/install.md:76]: these changes should be in a separate PR as they do not relate to its main topic

File: src/tests/golden/testdata/inflector/lib/Doctrine/Inflector/CachedWordInflector.php [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?php
 
-declare(strict_types=1);
+declare(strict_types = 1);
```
Comment (i582) [inline src/tests/golden/testdata/inflector/lib/Doctrine/Inflector/CachedWordInflector.php:3]: changes to test data formatting only pollute the PR, making it difficult to review

File: src/linter/worker.go [hunk 3]
```diff
@@ -315,6 +315,11 @@ func (w *Worker) analyzeFile(file *workspace.File, rootNode *ir.Root) (*rootWalk
 	}
 	walker.afterLeaveFile()
 
+	if !walker.declareSection {
+		walker.Report(rootNode, LevelWarning, "noDeclareSection", "You have no declare section with strict_types=1 argument")
```
Comment (i582) [inline src/linter/worker.go:319]: maybe "missed `declare(strict_types=1)` directive"?

File: docs/install.md (hunk 1 above)
Comment (Hidanio) [reply docs/install.md:76]: I am totally agree with you (and that was the original title of this PR), but we had an agreement with @Danil42Russia  that this PR would also include changes to the documentation, corrections to some warnings, and test data. But if you insist, I can separate them

File: src/tests/golden/testdata/inflector/lib/Doctrine/Inflector/CachedWordInflector.php (hunk 2 above)
Comment (Hidanio) [reply src/tests/golden/testdata/inflector/lib/Doctrine/Inflector/CachedWordInflector.php:3]: Reply is the same as to the previous comment

File: src/linter/worker.go (hunk 3 above)
Comment (Hidanio) [reply src/linter/worker.go]: I used a more formalized form, I agree, it could be simplified

File: docs/install.md (hunk 1 above)
Comment (i582) [reply docs/install.md:76]: ok!
